app/src/main/java/org/mozilla/focus/MainActivity.kt

```diff
@@ -48,7 +50,7 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener {
 
         setContentView(R.layout.activity_main)
 
-        sessionManager.handleIntent(this, intent, savedInstanceState)
+        IntentValidator.validateOnCreate(this, intent, savedInstanceState, ::onValidBrowserIntent)
```

Comment:
What is this :: notation?

Reply:
`::` is slightly overloaded (sigh, I know, I know...) but generally indicates access to meta information like `MainActivity::class` instead of `MainActivity.class` in Java. `::` without a receiver indicates an implicit `this` receiver, i.e. `this::`.

In our particular case, `::` is used to provide a function reference, e.g. `this::aFunctionOfTheClass`. Taking in a function is the same if it's a lambda or if it's a named function, e.g. in this contrived example:
```kotlin
fun isLessThanTwo(value: Int) = value < 2

fun main() {
    val values = listOf(1, 2, 3, 4)

    // These are equivalent!
    val namedFunctionReference = values.filter(::isLessThanTwo)
    val anonymousFunctionReference = values.filter { it < 2 }
}
```

Named function references are useful for improving readability of complex expressions, e.g. `values.filter(::isPrime)`, or that you don't want to duplicate the code for (like `onValidBrowserIntent`, here).